src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideLogsStep.groovy

```diff
@@ -0,0 +1,231 @@
+
+        private final long startTimeMillis
+        private final EnvVars envVars
+        private final String logDirPath
+        private final TaskListener listener
+        private RestApiClient apiClient
+
+        private final unsupportedProvideLogsMsg = "Downloading report folders is not supported for this ecu.test version. Please use ecu.test >= 2024.2 instead."
+
+        ExecutionCallable(long timeout, long startTimeMillis, EnvVars envVars, String logDirPath, TaskListener listener) {
+            super(timeout, listener)
+            this.startTimeMillis = startTimeMillis
+            this.envVars = envVars
+            this.logDirPath = logDirPath
+            this.listener = listener
+        }
+        /**
+         * Calls downloadReportFolder via the RestApiClient for all report ids.
+         * Then extracts and saves the ecu.test log out of the returned zip folder.
+         * Logs a warning if no reports were returned by ecu.test.
+         * @return List of strings containing the paths of the logs.
+         */
+        @Override
+        ArrayList<String> execute() throws IOException {
+            listener.logger.println("Providing ecu.test logs to jenkins.")
+            ArrayList<String> logFileNames = ["ecu.test_out.log", "ecu.test_err.log"]
+            ArrayList<String> logPaths = []
+            try {
+                RestApiClient apiClient = RestApiClientFactory.getRestApiClient(envVars.get('ET_API_HOSTNAME'), envVars.get('ET_API_PORT'))
+                if (apiClient instanceof RestApiClientV1) {
+                    throw new AbortException(unsupportedProvideLogsMsg)
+                }
+                apiClient = (RestApiClientV2) apiClient
+                List<ReportInfo> reports = apiClient.getAllReports()
+
+                if (reports == null || reports.isEmpty()) {
+                    return []
+                }
+                for (def report : reports) {
+                    String reportDir = report.reportDir.split('/').last()
+                    if (!checkReportFolderCreationDate(reportDir)) {
+                        listener.logger.println("[WARNING] ecu.test reports contains folder older than this run. Path: ${report.reportDir}")
+                    }
+                    File reportFolderZip = apiClient.downloadReportFolder(report.testReportId)
+                    List<String> extractedFiles = ZipUtil.extractAndSaveFromZip(reportFolderZip, logFileNames, "${logDirPath}/${reportDir}")
+                    if (extractedFiles.size() != logFileNames.size()) {
+                        listener.logger.println("[WARNING] ${report.reportDir} is missing one or all log files!")
+                    }
+                    logPaths.addAll(extractedFiles)
+                }
+            }
+            catch (ApiException e) {
+                if (e instanceof de.tracetronic.cxs.generated.et.client.v2.ApiException && e.code == 404) {
+                    throw new AbortException(unsupportedProvideLogsMsg)
```

Comment:
see ac's `pipeline status: unstable if feature is not available (ecu.test <2024.2)`